includes/plugins/woocommerce-subscriptions/class-subscriptions-tiers.php

```diff
@@ -121,37 +204,53 @@ private static function render_product_card( $product, $show_variation_attribute
 		} else {
 			$price = $product->get_price_html();
 		}
-		$product_name = $product->get_title();
-		if ( $product->is_type( 'variation' ) ) {
-			if ( $show_variation_attributes ) {
-				$product_name = sprintf(
-					'%s (%s)',
-					$product_name,
-					implode( ', ', $product->get_variation_attributes() )
-				);
-			}
-		}
 
 		?>
-		<label class="newspack-ui__input-card">
+		<label class="newspack-ui__input-card <?php echo $current ? esc_attr( 'current' ) : ''; ?>">
 			<?php if ( $current ) : ?>
 				<span class="newspack-ui__badge newspack-ui__badge--primary"><?php _e( 'Current', 'newspack-plugin' ); ?></span>
 			<?php endif; ?>
 			<input type="radio" name="product_id" value="<?php echo esc_attr( $product->get_id() ); ?>" <?php echo esc_attr( $selected ? 'checked' : '' ); ?>>
-			<strong><?php echo esc_html( $product_name ); ?></strong>
+			<strong><?php echo esc_html( self::get_product_title( $product, $show_variation_attributes ) ); ?></strong>
 			<span class="newspack-ui__helper-text"><?php echo $price; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?></span>
 		</label>
 		<?php
 	}
 
+	/**
+	 * Render existing subscription info.
+	 *
+	 * @param \WC_Product      $product      Product.
+	 * @param \WC_Subscription $subscription Subscription.
+	 */
+	public static function render_existing_subscription_info( $product, $subscription ) {
+		?>
+		<div class="newspack-ui__notice newspack-ui__notice--warning">
+			<span class="newspack-ui__notice__content">
+				<?php
+				printf(
+					/* translators: %s: subscription product name */
+					esc_html__( 'You already have %s active.', 'newspack-plugin' ),
```

Comment:
I think I saw some discussion between you and @thomasguillot on the wording of this message. Not to throw a wrench in, but what do you think about this instead?

```suggestion
					esc_html__( 'You already have an active subscription: %s.', 'newspack-plugin' ),
```

<img width="479" height="321" alt="Screenshot 2025-09-16 at 12 03 15 PM" src="https://github.com/user-attachments/assets/c1783e21-8f99-4448-92f7-891643c09e4b" />

Reply:
We avoided using the word subscription because publishers may label this differently (e.g., membership, plan, etc...).

In the future, I imagine a setting where publishers can define their own term, but for now I recommend keeping the warning message generic.

Reply:
Aha, thanks for the context!